crates/neon/src/lifecycle.rs

```diff
@@ -54,6 +58,36 @@ pub(crate) struct InstanceData {
 
     /// Shared `Channel` that is cloned to be returned by the `cx.channel()` method
     shared_channel: Channel,
+
+    /// Table of user-defined global cells.
+    globals: GlobalTable,
+}
+
+pub(crate) struct GlobalTable {
+    cells: Vec<Option<Box<dyn Any + Send>>>,
+}
+
+impl GlobalTable {
+    fn new() -> Self {
+        Self { cells: Vec::new() }
+    }
+}
+
+impl Index<usize> for GlobalTable {
+    type Output = Option<Box<dyn Any + Send>>;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.cells[index]
```

Comment:
Since this doesn't resize, it will panic if it hasn't been used yet. I don't think `.borrow(..)` panicking, but `borrow_mut(..)` not panicking is great UX.

Reply:
Yeah. It's just a `pub(crate)` internal data structure and this is an unused method (required because `IndexMut` extends `Index`) so I didn't worry too much about this.

I liked being able to use the `[]` syntax because I found it more readable, but it's only used once or twice so maybe it just doesn't pay for itself and I should just use an inherent method.